### PR TITLE
don't make changes to software installation directory when using --sanity-check-only

### LINKED ISSUE #3757: --sanity-check-only requires write access to software install dir
```
$ eb UCX-1.9.0-GCCcore-10.2.0.eb --sanity-check-only
...
ERROR: Traceback (most recent call last):
  File "/usr/lib64/python3.8/shutil.py", line 788, in move
    os.rename(src, real_dst)
PermissionError: [Errno 13] Permission denied: '/software/UCX/1.9.0-GCCcore-10.2.0/easybuild/reprod' -> '/software/UCX/1.9.0-GCCcore-10.2.0/easybuild/reprod_20210624104127_2701'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib64/python3.8/site-packages/easybuild/tools/filetools.py", line 2516, in move_file
    shutil.move(path, target_path)
  File "/usr/lib64/python3.8/shutil.py", line 798, in move
    copytree(src, real_dst, copy_function=copy_function,
  File "/usr/lib64/python3.8/shutil.py", line 554, in copytree
    return _copytree(entries=entries, src=src, dst=dst, symlinks=symlinks,
  File "/usr/lib64/python3.8/shutil.py", line 455, in _copytree
    os.makedirs(dst, exist_ok=dirs_exist_ok)
  File "usr/lib64/python3.8/os.py", line 223, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/software/UCX/1.9.0-GCCcore-10.2.0/easybuild/reprod_20210624104127_2701'
```

### PR DESCRIPTION
**Pull request by boegel**
fixes #3757

### CODE REVIEW

File: easybuild/framework/easyblock.py [hunk 1]
```diff
@@ -3695,6 +3695,11 @@ def build_and_install_one(ecdict, init_env):
                 new_log_dir = os.path.join(app.builddir, config.log_path(ec=app.cfg))
             else:
                 new_log_dir = os.path.dirname(app.logfile)
+
+        # if we're only running the sanity check, we should copy anything new to the installation directory
```
Comment (akesandgren) [inline easybuild/framework/easyblock.py:3699]: Comment is opposite to what is done, i.e. s/we should copy/we shouldn't copy/

File: test/framework/options.py [hunk 2]
```diff
@@ -5929,6 +5928,15 @@ def test_sanity_check_only(self):
         outtxt = self.eb_main(args + ['--skip-extensions'], do_build=True, raise_error=True)
         self.assertTrue("Sanity check for toy successful" in outtxt)
 
+        # restore fail, we want a passing sanity check for the next check
+        move_file(libbarbar + '.moved', libbarbar)
+
+        # check use of --sanity-check-only when installation directory is read-only;
+        # cfr. https://github.com/easybuilders/easybuild-framework/issues/3757
+        adjust_permissions(ebroottoy, stat.S_IWUSR, add=False, recursive=True)
```
Comment (akesandgren) [inline test/framework/options.py:5936]: Unless guaranteed to be so already, you need to remove group (and other) write permissions too.

**Review by akesandgren — CHANGES_REQUESTED** [review]

File: test/framework/options.py (hunk 2 above)
Comment (boegel) [reply test/framework/options.py:5936]: OK, doesn't hurt to be a bit more strict here...

**Review by akesandgren — APPROVED** [review]
LGTM

**Pull request merged**